src/Compilers/CSharp/Portable/Binder/Semantics/BestTypeInferrer.cs

```diff
@@ -258,23 +249,35 @@ private static TypeSymbol Better(TypeSymbol type1, TypeSymbol type2, Conversions
 
                 if (type1.Equals(type2, TypeCompareKind.IgnoreDynamicAndTupleNames))
                 {
-                    return MethodTypeInferrer.Merge(type1, type2, conversions.CorLibrary);
+                    (bool _, TypeSymbol merged) = MethodTypeInferrer.TryMerge(type1, type2, VarianceKind.Out, conversions);
```

Comment:
>VarianceKind.Out [](start = 92, length = 16)

Variance = Out makes sense for lambda returns, but it feels wrong for implicitly-typed arrays.
Also, ternary/conditional expressions that are `ref` might have an issue.

Reply:
Logged #30432 and added tests for conditional operators and lambdas.

---
In reply to: [223838549](https://github.com/dotnet/roslyn/pull/29727#discussion_r223838549) [](ancestors = 223838549)